lms/djangoapps/grades/models.py

```diff
@@ -0,0 +1,46 @@
+from django.contrib.auth.models import User
+from django.db import models
+from model_utils.models import TimeStampedModel
+
+from xmodule_django.models import BlockTypeKeyField, CourseKeyField, UsageKeyField
+
+class PersistentSubsectionGradeModel(TimeStampedModel):
+    """
+    A django model tracking persistent grades at the subsection level.
+
+    TODO: intended query patterns are fluid at the moment, document them here as they're firmed up.
+    """
+
+    class Meta(object):
+        index_together = [
+            # TODO: nail down indices as we flesh out the API layer
+        ]
+
+        unique_together = (('user_id', 'course_id', 'block_type', 'block_id'))
+
+
+    subtree_edited_date = models.DateTimeField('last content edit timestamp')
+    user = models.ForeignKey(User)  # CAUTION: FK may not exist in the future
+    earned_grade = models.IntegerField()
+    max_grade = models.IntegerField()
+    visible_blocks = models.ForeignKey(BlockRecord)
+
+    # TODO: Make sure this matches up w/ what Nimisha's exposing in her PR
+    course_version = models.CharField('guid of latest course version', max_length=255)
+
+    # These 3 are essentially a deconstructed UsageKey
+    course_id = CourseKeyField(max_length=255)
+    block_type = BlockTypeKeyField(max_length=255)
+    block_id = models.CharField(max_length=255)  # TODO: CharField correct here?
```

Comment:
Advantages to storing these as separate fields:
1. can easily query by `block_type`.
2. eliminate redundant `course_id` information within a `block_usage_key` - less space usage.

Advantages to storing these as simply a `UsageKeyField`:
1. in case block-usage-keys change in the future and there are more components within keys, this table will continue to support it assuming the `UsageKeyField` serializes it correctly.
2. consistent with the `StudentModule` data model as it stores `course_id` and `module_state_key` (a usage key).